tests/e2e/runtime.go

```diff
@@ -0,0 +1,465 @@
+}
+
+func (pk *pkWrapper) String() string {
+	return pk.pk.String()
+}
+
+func (pk *pkWrapper) Equal(other signature.PublicKey) bool {
+	// Can't compare bytes, because signature.PublicKey can't be sliced...
+	return pk.String() == other.String()
+}
+
+func (pk *pkWrapper) Verify(context, message, signature []byte) bool {
+	return pk.pk.Verify(coreSig.Context(context), message, signature)
+}
+
+// Also really annoying:
+//
+// cannot use signer (type "github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Signer) as type "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".Signer in argument to kvInsert:
+//	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Signer does not implement "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".Signer (wrong type for ContextSign method)
+//		have ContextSign("github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Context, []byte) ([]byte, error)
+//		want ContextSign([]byte, []byte) ([]byte, error)
+type signerWrapper struct {
+	signer coreSig.Signer
+}
+
+func newSignerWrapper(signer coreSig.Signer) *signerWrapper {
+	return &signerWrapper{signer: signer}
+}
+
+func (sw *signerWrapper) Public() signature.PublicKey {
+	return newPublicKeyWrapper(sw.signer.Public())
+}
+
+func (sw *signerWrapper) ContextSign(context, message []byte) ([]byte, error) {
+	return sw.signer.ContextSign(coreSig.Context(context), message)
+}
+
+func (sw *signerWrapper) String() string {
+	return sw.signer.String()
+}
+
+func (sw *signerWrapper) Reset() {
+	sw.signer.Reset()
+}
+
+type kvKey struct {
+	Key string `json:"key"`
+}
+
+type kvKeyValue struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+func kvInsert(rtc client.RuntimeClient, signer signature.Signer, nonce uint64, key string, value string) error {
+	ctx := context.Background()
+
+	tx := types.NewTransaction(nil, "Insert", kvKeyValue{
+		Key:   key,
+		Value: value,
+	})
+	tx.AppendSignerInfo(signer.Public(), nonce)
+	stx := tx.PrepareForSigning()
+	stx.AppendSign(signer)
+
+	if _, err := rtc.SubmitTx(ctx, stx.UnverifiedTransaction()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kvRemove(rtc client.RuntimeClient, signer signature.Signer, nonce uint64, key string) error {
+	ctx := context.Background()
+
+	tx := types.NewTransaction(nil, "Remove", kvKey{
+		Key: key,
+	})
+	tx.AppendSignerInfo(signer.Public(), nonce)
+	stx := tx.PrepareForSigning()
+	stx.AppendSign(signer)
+
+	if _, err := rtc.SubmitTx(ctx, stx.UnverifiedTransaction()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kvGet(rtc client.RuntimeClient, key string) (string, error) {
+	ctx := context.Background()
+
+	var resp kvKeyValue
+	if err := rtc.Query(ctx, client.RoundLatest, "Get", kvKey{Key: key}, &resp); err != nil {
+		return "", err
+	}
+	return resp.Value, nil
+}
+
+func SimpleKVTest(net *oasis.Network, rtc client.RuntimeClient) error {
+	signer := newSignerWrapper(net.Entities()[1].Signer())
```

Comment:
You can use the predefined SDK test keys:
https://github.com/oasisprotocol/oasis-sdk/blob/main/client-sdk/go/testing/testing.go